app/src/pages/playground/InvocationParametersFormFields.tsx

```diff
@@ -198,144 +235,109 @@ export const InvocationParametersFormFields = ({
-          updateInstanceModelInvocationParameters({
-            instanceId: instance.id,
-            invocationParameters: instance.model.invocationParameters.filter(
-              (p) => !areInvocationParamsEqual(p, field)
-            ),
-          });
-        }
-        return;
+      if (!field.invocationName) {
+        throw new Error("Invocation name is required");
       }
-
-      if (existingParameter) {
-        const input = makeInvocationParameterInput(field, value);
-        if (input) {
-          updateInstanceModelInvocationParameters({
-            instanceId: instance.id,
-            invocationParameters: instance.model.invocationParameters.map(
-              (p) => (areInvocationParamsEqual(p, field) ? input : p)
-            ),
-          });
-        }
+      if (value === undefined) {
+        deleteInvocationParameterInput({
+          instanceId,
+          invocationParameterInputInvocationName: field.invocationName,
+        });
       } else {
-        const input = makeInvocationParameterInput(field, value);
-        if (input) {
-          updateInstanceModelInvocationParameters({
-            instanceId: instance.id,
-            invocationParameters: [
-              ...instance.model.invocationParameters,
-              input,
-            ],
-          });
-        }
+        upsertInvocationParameterInput({
+          instanceId,
+          invocationParameterInput: makeInvocationParameterInput(field, value),
+        });
       }
     },
-    [instance, updateInstanceModelInvocationParameters]
+    [instanceId, upsertInvocationParameterInput, deleteInvocationParameterInput]
   );
 
-  // It is safe to render this component if the model name is not set for non-azure models
-  // Hooks will still run to filter invocation parameters to only include those supported by the model
-  // but no form fields will be rendered if the model name is not set
-  if (model.modelName === null && model.provider !== "AZURE_OPENAI") {
+  // Reduce our invocation parameters array into a form state object
+  const values = useMemo(() => {
+    return supportedInvocationParameterDefinitions
+      .filter(
+        (field) =>
+          // remove parameters that we want to ignore in the form
+          !(
+            field.canonicalName != null &&
+            paramsToIgnoreInInvocationParametersForm.includes(
+              field.canonicalName
+            )
+          )
+      )
+      .map((field) => {
+        const existingParameter = instanceInvocationParameters.find((p) =>
+          areInvocationParamsEqual(p, field)
+        );
+        const value = existingParameter
+          ? getInvocationParameterValue(field, existingParameter)
+          : undefined;
+        return {
+          [field.invocationName!]: value ?? null,
+        };
+      })
+      .reduce(
+        (acc, param) => {
+          return { ...acc, ...param };
+        },
+        {} as Record<string, unknown>
+      );
```

Comment:
do we need this map then reduce? seem slike this can just be done in a single reduce

Reply:
Fixed